### DISCUSSION TITLE
Setup / Nvidia GPU

### DISCUSSION THREAD

**Discussion by Cah1982** [blakeblackshear/frigate#7566] [Ask A Question] [answered]
So I have just got a container in docker running frigate, but I have a few questons

I currently have GPU working with the below

`ffmpeg:
  hwaccel_args: preset-nvidia-h264`

But how do I get this working for detectors, the documentation mentions tensorrt but i cant get my head around how this works

at minute i have in the config 

`detectors:
  cpu1:
    type: cpu
    num_threads: 3
    model:
      path: "/custom_model.tflite"
  cpu2:
    type: cpu
    num_threads: 3`

But when I add zones (generated from the gui) it says the config isnt correct

**Comment by NickM-27** [comment]
Seems like you're asking 2 different questions here

> But how do I get this working for detectors, the documentation mentions tensorrt but i cant get my head around how this works

tensorrt setup is outlined here: https://docs.frigate.video/configuration/detectors#nvidia-tensorrt-detector

if you have specific questions from there that will be easier to answer

> But when I add zones (generated from the gui) it says the config isnt correct

This is unrelated to tensorrt, it fails likely because the spacing or format is wrong, the config editor will tell you what is wrong

**Comment by Cah1982** [comment]
yeah its taken me hours to stich the code correctly with right tabs, and in right order

So with this tensorrt, i have read that page, but im quite new to docker / linux systems

nvidia-container-runtime is working and shows when the GPU is being used

Its just where do I go from there to configure it 

`mkdir trt-models
wget https://github.com/blakeblackshear/frigate/raw/master/docker/tensorrt_models.sh
chmod +x tensorrt_models.sh
docker run --gpus=all --rm -it -v `pwd`/trt-models:/tensorrt_models -v `pwd`/tensorrt_models.sh:/tensorrt_models.sh nvcr.io/nvidia/tensorrt:22.07-py3 /tensorrt_models.sh`

I have seen the above, but as i didnt understand it, so havent run this

**Comment by Cah1982** [comment]
Ok so i got it working, but how do you know when new images of the below are released?

ghcr.io/blakeblackshear/frigate:0.12.0-beta10-tensorrt

**Reply by NickM-27** [reply, marked as answer]
You should be using 

ghcr.io/blakeblackshear/frigate:stable-tensorrt

**Reply by Cah1982** [reply]
cheers

in the GUI how would i fix this, but everything seems to be working

VAINFO OUTPUT
Return Code: 3

Process Error:

error: XDG_RUNTIME_DIR is invalid or not set in the environment.
error: can't connect to X server!
libva info: VA-API version 1.17.0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/nvidia_drv_video.so
libva info: va_openDriver() returns -1
vaInitialize failed with error code -1 (unknown libva error),exit

**Reply by NickM-27** [reply]
vainfo does not apply to nvidia, nothing is wrong

**Reply by Cah1982** [reply]
cheers, final question

![image](https://github.com/blakeblackshear/frigate/assets/44786907/1ca01eb8-2917-496d-9234-67fae0dc2fae)


does that look correct for usages to suggest the GPU is doing the work and not the CPU on last 3 items

**Reply by NickM-27** [reply]
Yes, there is still work that has to be done on the CPU as well as the GPU

**Reply by Cah1982** [reply]
cheers I didnt know you could setup a sub stream for the monitoring side, that reduced load on system

![image](https://github.com/blakeblackshear/frigate/assets/44786907/13cbe694-8196-4f08-8a95-2623fce4386e)
